app/models/service_template_provision_request.rb

```diff
@@ -43,6 +43,7 @@ def my_role(action = nil)
   end
 
   def my_zone
+    @my_zone ||= options.fetch_path(:dialog, "dialog_zone") || service_template.my_zone
```

Comment:
Why is this the only one of these being memoized? While it's unlikely to change, I don't think any of the other `#my_zone` methods in the app are doing this.

Reply:
The zone of a request, specified by the dialog or the service template, is set once the request is submitted. And ```my_zone``` would be called multiple times during the request process.

Reply:
The dialog field name `zone` is not very unique and could be created by a dialog editor to reference something other than MIQ zones.  I'd suggest that if the zone name is coming from the dialog we validate that it is actually an existing zone in MIQ, otherwise work will get queued and never processed.

Create a separate method for that validation and have it return `nil` if there is no field, or the field reference something that is not an existing MIQ zone:

You would end up with something like: `@my_zone ||= dialog_zone || service_template.my_zone`